copy/copy.go

```diff
@@ -703,22 +702,17 @@ func checkImageDestinationForCurrentRuntime(ctx context.Context, sys *types.Syst
 		if err != nil {
 			return errors.Wrapf(err, "Error parsing image configuration")
 		}
-
-		wantedOS := runtime.GOOS
-		if sys != nil && sys.OSChoice != "" {
-			wantedOS = sys.OSChoice
-		}
-		if wantedOS != c.OS {
-			return fmt.Errorf("Image operating system mismatch: image uses %q, expecting %q", c.OS, wantedOS)
+		wantedPlatform, err := manifest.WantedPlatform(sys)
```

Comment:
I don’t think a single value returned here is going to work long-term (a new variant of an architecture may support several previous variants).

That’s not a strong reason not to do it this way _for now_ at all, just a reason not to make it a public API (see below).

Reply:
Ah, I see where you are going with this. It is probably a good place to put the "fallback" variant logic inside the platform matcher rather than in the image selection process itself.